### ISSUE TITLE
wsltty terminal closes as soon as it opens

### ISSUE THREAD

**Issue by Luiscri** [mintty/wsltty#186]
I'm experimenting this type of error. Reading other issues here I launched it with the hold option and it showed this error message:

```
/bin/wslbridge: Exit 1.
error: 'C:\Users\luisc\AppData\Local\wsltty\bin\wslbridge-backend' backend program is missing
```

I installed WSL following the instructions of Microsoft official page and did not configure anything else. I'm pretty new to working on Windows, so sorry if I don't understand or know how to do quite simple things.

**Comment by Biswa96** [comment]
Did you check if that `"C:\Users\luisc\AppData\Local\wsltty\bin\wslbridge-backend"` file exists?

**Comment by Luiscri** [comment]
No it doesn't, it only exists one called `wslbridge` under the same directory. How can I generate the missing file?

**Comment by Biswa96** [comment]
Try to run the wsltty installer or the easiest way to extract the backend file from the wsltty installer with 7zip (or equivalent).

**Comment by Luiscri** [comment]
Running again the installer solves the problem, but after certain time the error appears again.
Now it also says:

```
note: backend error output: -v: -c: line 0: unexpected EOF while looking for matching `''
-v: -c: line 1: syntax error: unexpected end of file
```

**Comment by Biswa96** [comment]
What is Windows 10 version? I think I've seen this error message in another issue...

**Comment by OlivierMary** [comment]
@Luiscri  WSL 2 ?

**Comment by Biswa96** [comment]
If you are using Windows 10 insider build fast preview 18945 or greater there is a chance to get that error. Here is the same error occurs with other user https://github.com/rprichard/wslbridge/issues/44#issuecomment-516364864. The workaround is to copy the `System32/bash.exe` from older insider build. Or use [wslbridge2](https://github.com/Biswa96/wslbridge2).

**Comment by OlivierMary** [comment]
> 
> 
> If you are using Windows 10 insider build fast preview 18945 or greater there is a chance to get that error. Here is the same error occurs with other user [rprichard/wslbridge#44 (comment)](https://github.com/rprichard/wslbridge/issues/44#issuecomment-516364864). The workaround is to copy the `System32/bash.exe` from older insider build. Or use [wslbridge2](https://github.com/Biswa96/wslbridge2).

wslbridge2 + wsltty ? How to process ?

**Comment by Biswa96** [comment]
wslbridge2 does not match in wsltty repository right now because there is a command line option (distribution name vs distribution GUID) that does not match. But one can execute [wslbridge2](https://github.com/Biswa96/wslbridge2/releases) binary in mintty as a regular command.

**Comment by OlivierMary** [comment]
It's the system used by https://github.com/microsoft/terminal ? I can run on it, but mouse interaction with tmux didn't works

**Comment by Biswa96** [comment]
Don't know.

**Comment by cesarga** [comment]
> 
> 
> If you are using Windows 10 insider build fast preview 18945 or greater there is a chance to get that error. Here is the same error occurs with other user [rprichard/wslbridge#44 (comment)](https://github.com/rprichard/wslbridge/issues/44#issuecomment-516364864). The workaround is to copy the `System32/bash.exe` from older insider build. Or use [wslbridge2](https://github.com/Biswa96/wslbridge2).

Thanks, using wslbridge2 worked.

I extracted the latest [release](https://github.com/Biswa96/wslbridge2/releases) to the wsltty bin directory and used these .bat to launch the terminals.

WSL1 distro:
```bat
@echo off
C:\Users\cesar\AppData\Local\wsltty\bin\mintty.exe -i "C:\Users\cesar\AppData\Local\wsltty\wsl.ico" --configdir="C:\Users\cesar\AppData\Roaming\wsltty" C:\Users\cesar\AppData\Local\wsltty\bin\wslbridge2.exe -d Ubuntu-16.04
```

WSL2 distro:
```bat
@echo off
C:\Users\cesar\AppData\Local\wsltty\bin\mintty.exe -i "C:\Users\cesar\AppData\Local\wsltty\wsl.ico" --configdir="C:\Users\cesar\AppData\Roaming\wsltty" C:\Users\cesar\AppData\Local\wsltty\bin\hvpty.exe -d Ubuntu-18.04
```

Just remember to replace the username.

**Comment by mintty** [comment]
Thanks. You seem to have edited the comment after first posting and replaced `hvpty` with `rawpty` for the WSL 2 case. Note that this will obstruct full transparency of mintty features, only those that are also supported by the MS terminal concept are available. Using `hvpty` gives you full mintty functionality.

The next mintty release shall dispatch to either `wslbridge2` or `hvpty` automatically when triggered with option `--WSL`.

**Comment by OlivierMary** [comment]
For generic ` .bat `

```bat
@echo off
rem Start mintty terminal for WSL package Ubuntu-18.04 in current directory
%LOCALAPPDATA%\wsltty\bin\mintty.exe -i "%LOCALAPPDATA%\wsltty\wsl.ico" --configdir="%APPDATA%\wsltty" %LOCALAPPDATA%\wsltty\bin\rawpty.exe wsl -d Ubuntu-18.04
```

**Comment by mintty** [comment]
See my previous comment. Using `rawpty`, you don't really have mintty anymore.
Next wsltty release shall solve all issues.

**Comment by AiOO** [comment]
Thanks to @cesarga and @OlivierMary's comments, I've succeed to run WSL on mintty. 😻

On the other hand, I think the `-d` option is not working because default distribution always runs regardless of its value.

**Comment by OlivierMary** [comment]
Only `rawpty` work for me, but mouse interactions with tmux does not work....



> 
> 
> See my previous comment. Using `rawpty`, you don't really have mintty anymore.
> Next wsltty release shall solve all issues.

I just configure font + colors -_-"

**Comment by alekseymvt** [comment]
@mintty 
> Using hvpty gives you full mintty functionality.

`
@echo off
rem WSL2 distro
%LOCALAPPDATA%\wsltty\bin\mintty.exe -i "%LOCALAPPDATA%\wsltty\wsl.ico" --configdir="%APPDATA%\wsltty" %LOCALAPPDATA%\wsltty\bin\hvpty.exe -d Ubuntu-18.04
`
Right? 
Windows 1903 (18980.1). WSL2 works. Mouse/keyboard works. Thanks.

**Comment by Biswa96** [comment]
There is an issue in hvpty with Alpine. Use it cautiously.

**Comment by dxhisboy** [comment]
To know what is happening, maybe add `-h always` to the mintty's argument.

**Comment by mintty** [comment]
Released 3.0.5 with new gateway.
